treeherder/model/models.py

```diff
@@ -1033,6 +1034,28 @@ def to_dict(self):
             'modified': self.modified,
         }
 
+    def to_mozlog_format(self):
+        """Convert a FailureLine into a mozlog formatted dictionary."""
+        data = {
+            "action": self.action,
+            "line_number": self.line,
+            "test": self.test,
```

Comment:
The new implementation will raise `AttributeError` if some of these keys don't exist. Are we sure all attributes are expected to exist? I seem to remember some are either optional or else new so not present in test runs on older branches. James would likely have a better idea :-)

Reply:
`self` is a model instance here so all attributes will exist but can have a value of `None`.  These keys are removed by https://github.com/mozilla/treeherder/pull/3991/files/95a2a2f83e679b56021180b6dbfc9564a339ce99#diff-e432623caa985e5ac4d48814de254a58R1055.

Reply:
Ah true - sorry! :-)